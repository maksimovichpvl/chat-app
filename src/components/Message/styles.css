.message {
  display: flex;
  margin: 10px 0; }
  .message--my {
    justify-content: flex-end;
    text-align: right; }
  .message--received {
    justify-content: flex-start;
    text-align: left; }
  .message__wrapper {
    word-wrap: break-word;
    max-width: 140px; }
  .message__username {
    font-weight: bold; }
  .message__msg {
    padding: 10px;
    border-radius: 5px; }
    .message__msg--my {
      background-color: #ccebec; }
    .message__msg--received {
      background-color: #e2edd9; }
