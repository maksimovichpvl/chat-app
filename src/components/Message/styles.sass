
.message
  display: flex
  margin: 10px 0
  &--my
    justify-content: flex-end
    text-align: right 
  &--received
    justify-content: flex-start
    text-align: left

  &__wrapper
    word-wrap: break-word
    max-width: 140px

  &__username
    font-weight: bold

  &__msg 
    padding: 10px
    border-radius: 5px
    &--my
      background-color: #ccebec
    &--received
      background-color: #e2edd9
