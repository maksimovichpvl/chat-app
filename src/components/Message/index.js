import React from 'react';
import PropTypes from 'prop-types';

import './styles.css';

const addSpecialStyles = (message, username) => 
  message.username === username? "--my" : "--received";

const Message = ({ message, username }) => (
  <div className={`message message${addSpecialStyles(message, username)}`}>
    <div className="message__wrapper">
      <div className={`message__username message${addSpecialStyles(message, username)}`}>
        {message.username}
      </div>
      <div className={`message__msg message__msg${addSpecialStyles(message, username)}`}>
        {message.content}
      </div>
      <div>{message.createdAt}</div>
    </div>
    </div>
);

Message.propTypes = {
  message: PropTypes.object.isRequired,
  username: PropTypes.string.isRequired
};

export default Message;