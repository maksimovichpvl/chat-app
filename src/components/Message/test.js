import React from 'react';
import renderer from 'react-test-renderer';

import Message from './index.js';
import { mount } from 'enzyme';

describe('Message Component', function() {
  const usernameMock = "Test username";
  const message = {
    username: usernameMock
  };

  it('renders without crashing', () => {
    const tree = renderer
      .create(<Message  username={usernameMock} message={message} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Correctly props', () => {
    const wrapper = mount(<Message username={usernameMock} message={message} />);
    expect(wrapper.props().username).toEqual(usernameMock);
    expect(wrapper.props().message).toEqual(message);
  });

  it('Correctly classNames', () => {
    const wrapper = mount(<Message username={usernameMock} message={message} />);
    expect(wrapper.find('.message--my').length).toEqual(2);
  });
})


