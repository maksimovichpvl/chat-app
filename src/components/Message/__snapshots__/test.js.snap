// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component renders without crashing 1`] = `
<div
  className="message message--my"
>
  <div
    className="message__wrapper"
  >
    <div
      className="message__username message--my"
    >
      Test username
    </div>
    <div
      className="message__msg message__msg--my"
    />
    <div />
  </div>
</div>
`;
